tests/js/client/shell/shell-index.js

```diff
@@ -0,0 +1,116 @@
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Dr. Frank Celler, Lucas Dohmen
+/// @author Copyright 2012, triAGENS GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const internal = require("internal");
+const arango = internal.arango;
+const errors = internal.errors;
+const print = internal.print;
+const { helper, versionHas } = require("@arangodb/test-helper");
+const platform = require('internal').platform;
+
+
+const cn = "UnitTestsCollection";
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite: basics
+////////////////////////////////////////////////////////////////////////////////
+
+function IndexSuite() {
+  'use strict';
+
+  let collection = null;
+
+  return {
+
+    setUp : function () {
+      internal.db._drop(cn);
+      collection = internal.db._create(cn);
+    },
+
+    tearDown : function () {
+      internal.db._drop(cn);
+      collection = null;
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test: indexes
+////////////////////////////////////////////////////////////////////////////////
+
+    testIndexProgress : function () {
+      var c = internal.db._create("c0l", {numberOfShards: 3, replicationFactor: 2});
+      var docs = [];
+      const sleep = require('internal').sleep;
+      for (var j = 0; j < 16; ++j) {
+        docs=[];
+        for (var i = 0; i < 10240; ++i) {
+          docs.push({name : "name" + i});
+        }
+        c.insert(docs);
+      }
+      
+      internal.debugSetFailAt("fillIndex::pause");
+      var idx = { name:"progress", type: "persistent", fields:["name"], inBackground: true};
+      arango.GET(`/_api/index?collection=c0l`);
+
+      var job = arango.POST_RAW(`/_api/index?collection=c0l`, idx,
+                                {"x-arango-async": "store"}).headers["x-arango-async-id"];
+      let count = 0;
+      var progress = 0.0;
+      var idxs = [];
+      while (true) {
+        idxs = arango.GET(`/_api/index?collection=c0l&withHidden=true`).indexes;
+        if (idxs.length > 1) {
+          break;
+        }
+        sleep(0.1);
+      }
+
+      while (true) {
+        idx = arango.GET(`/_api/index?collection=c0l&withHidden=true`).indexes[1];
+        internal.debugSetFailAt("fillIndex::unpause");
+        if (idx.hasOwnProperty("progress") && idx.progress < 100.0) {
+          assertTrue(idx.progress >= progress);
+          progress = idx.progress;
+          print(progress.toPrecision(4) + "%");
+          sleep(0.1);
+        } else {
+          break;
+        }
+      }
+
+      internal.debugRemoveFailAt("fillIndex::pause");
```

Comment:
These two should go into a finally clause, otherwise they could remain behind if an exception is thrown.